### DISCUSSION TITLE
BT-722-Contract mismatch definition ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#411] [Q&A] [answered]
Hi Lab
in 25.json, 29.json 
{
            "id" : "BT-722-Contract",
            "contentType" : "field",
            **"displayType" : "TEXTAREA",**
            "description" : "Contract EU Funds Programme",
            "_label" : "field|name|BT-722-Contract"
          }

in fields.json
"id" : "BT-722-Contract",
    "parentNodeId" : "ND-ContractEUFunds",
    "name" : "Contract EU Funds Programme",
    "btId" : "BT-722",
    "xpathAbsolute" : "/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:SettledContract/efac:Funding/cbc:FundingProgramCode",
    "xpathRelative" : "cbc:FundingProgramCode",
    "xsdSequenceOrder" : [ { "cbc:FundingProgramCode" : 2 } ],
    **"type" : "code",**
    "legalType" : "CODE",
    
 "sdkVersion" : "1.6.0"

Bye
Renato

**Comment by YvesJo** [comment]
I guess this UI display choice has been elected for usability purpose so that the user may type some letters and get an extract of possible values from the long codelist for which a dropdown would not be appropriate.

**Reply by rouschr** [reply]
Yes this looks similar and had been answered below.

**Comment by rengit62** [comment]
Yes Yves, the nice effect that you wrote is done in Notice Editor by the tom-select.js.
But with or without tom-select.is,  with the current definition of "BT-722-Contract" there is an error as in jpg ErrorNoticeType25_2.jpg, and the combo is not populated with codelist value , jpg ErrorNoticeType25_1.jpg
![ErrorNoticeType25_1](https://user-images.githubusercontent.com/52211037/233115678-428c45b5-7b8f-45d0-927c-a13075859407.jpg)
![ErrorNoticeType25_2](https://user-images.githubusercontent.com/52211037/233115690-bab181c7-2966-40d9-8353-dc81340f7b7e.jpg)

Instead setting 
 "id" : "BT-722-Contract",
            "contentType" : "field",
            "displayType" : "COMBOBOX",
            "description" : "Contract EU Funds Programme",
            "_label" : "field|name|BT-722-Contract"
          }

we get the list : 
![NoticeType25_OK](https://user-images.githubusercontent.com/52211037/233118465-3dbde09a-9d00-4d0d-923c-b2bf9a6e7b9c.jpg)
besides that, BT-262 is a very very long list of supplies, services,  and is defines as 
"id" : "BT-262-Lot",
            "contentType" : "field",
            "displayType" : "COMBOBOX",
            "description" : "Main Classification Code",
            "_label" : "field|name|BT-262-Lot"

![NoticeType25_MainClassification](https://user-images.githubusercontent.com/52211037/233119834-510262d0-f963-423f-adfb-e52791443650.jpg)

Bye !
Renato

**Reply by YvesJo** [reply]
Thanks Renato for these details.
I will raise this to my colleagues for them to look at it.

**Comment by rouschr** [comment, marked as answer]
The message with:
`TypeError: this.formElement.populate is not a function at CodeInputField.populate (notice-type-definition.js:457:22)`
 looks like a bug.
For the field "BT-722-Contract" (of type code) the codelist is "eu-programme".
In the NTD 25 we have:
```
    "id" : "BT-722-Contract",
            "contentType" : "field",
            "displayType" : "TEXTAREA",
            "description" : "Contract EU Funds Programme",
            "_label" : "field|name|BT-722-Contract"
```
TEXTAREA
So here the NTD (notice-types definition data) is faulty, and as we are now aware of this, we will change this and similar cases from TEXTAREA to COMBOBOX to avoid this problem.
Probably something to change for the notice-types data of SDK 1.8.0.
This could also be something we could automatically detect earlier.
